tests/test-maximize.lua

```diff
@@ -142,6 +153,90 @@ local steps = {
         assert(geo_to_str(original_geo) == geo_to_str(new_geo),
             geo_to_str(original_geo) .. " == " .. geo_to_str(new_geo))
 
+        c:kill()
+
+        return true
+    end,
+    -- Now, start some clients maximized
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_before=true})
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if not c then return end
+
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical)
+        assert(c.maximized)
+
+        c:kill()
+
+        return true
+    end,
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_after=true})
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if not c then return end
+
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical)
+
+        -- There is a small 1 frame delay to lower the risk of a race condition
+        if c.maximized then
+            c:kill()
+            return true
+        end
+    end,
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_after=true})
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if not c then return end
+
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical)
+
+        -- It might happen in the second try
+        if not c.maximized then return end
+
+        c:kill()
+
+        client.disconnect_signal("request::geometry", awful.ewmh.geometry)
+        client.disconnect_signal("request::geometry", awful.ewmh.merge_maximization)
+        client.connect_signal("request::geometry", geometry_handler)
```

Comment:
Can/should this "change some signals"-part be moved into the next function to keep the distinction between "what is a single test" clearer?